moveit_ros/planning/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp

```diff
@@ -369,52 +387,66 @@ robot_model::SolverAllocatorFn KinematicsPluginLoader::getLoaderFunction(const s
-                                             "deprecated in favor of kinematics_solver_ik_links "
-                                             "rosparam array.");
+              ksolver_ik_link = node_->get_parameter(ksolver_ik_link_param_name).get_value<std::string>();
+              RCLCPP_WARN(LOGGER_KINEMATICS_PLUGIN_LOADER, "Using kinematics_solver_ik_link rosparam is "
+                                                           "deprecated in favor of kinematics_solver_ik_links "
+                                                           "rosparam array.");
               iksolver_to_tip_links[known_groups[i].name_].push_back(ksolver_ik_link);
             }
           }
 
           // Allow a kinematic solver's tip links to be specified on the rosparam server as an array
           std::string ksolver_ik_links_param_name;
-          if (nh.searchParam(base_param_name + "/kinematics_solver_ik_links", ksolver_ik_links_param_name))
+          if (ksolver_params->has_parameter(base_param_name + "/kinematics_solver_ik_links"))
           {
-            XmlRpc::XmlRpcValue ksolver_ik_links;
-            if (nh.getParam(ksolver_ik_links_param_name, ksolver_ik_links))  // has custom rosparam-based tip link(s)
+            ksolver_ik_links_param_name =
+                node_->get_parameter(base_param_name + "/kinematics_solver_ik_links").get_value<std::string>();
+            // XmlRpc::XmlRpcValue ksolver_ik_links;
+            std::vector<std::string> ksolver_ik_links;
+            if (ksolver_params->has_parameter(ksolver_ik_links_param_name))
             {
-              if (ksolver_ik_links.getType() != XmlRpc::XmlRpcValue::TypeArray)
+              ksolver_ik_links =
+                  node_->get_parameter(ksolver_ik_links_param_name).get_value<std::vector<std::string>>();
+              if (node_->get_parameter(ksolver_ik_links_param_name).get_type_name().find("array"))
               {
-                ROS_WARN_STREAM_NAMED(LOGNAME, "rosparam '" << ksolver_ik_links_param_name << "' should be an XmlRpc "
-                                                                                              "value type 'Array'");
+                RCLCPP_WARN(LOGGER_KINEMATICS_PLUGIN_LOADER, "the parameter '%s' needs to be of type 'Array'",
+                            ksolver_ik_links_param_name.c_str());
               }
               else
               {
                 for (int32_t j = 0; j < ksolver_ik_links.size(); ++j)
                 {
-                  ROS_ASSERT(ksolver_ik_links[j].getType() == XmlRpc::XmlRpcValue::TypeString);
-                  ROS_DEBUG_STREAM_NAMED(LOGNAME, "found tip " << static_cast<std::string>(ksolver_ik_links[j])
-                                                               << " for group " << known_groups[i].name_);
+                  // ROS_ASSERT(ksolver_ik_links[j].getType() == XmlRpc::XmlRpcValue::TypeString);
```

Comment:
I think we could just use the parameter type `PARAMETER_STRING_ARRAY` to implement this use case.